raven_framework

```diff
@@ -31,34 +31,49 @@ for var in "$@"; do
   ARGS[${#ARGS[@]}]="$var"
 done
 
-# run Driver.py based on the mode chosen
-if [ "$MODE" = 'help' ]; then
-  echo ''
-  echo 'Usage:'
-  echo '  raven_framework [-D][interfaceCheck][interactive][interactiveCheck] file'
-  echo ''
-  echo 'Description:'
-  echo '  Runs RAVEN using the input file given by "file".'
-  echo ''
-  echo 'Options:'
-  echo '  -D'
-  echo '    Development mode.  Turns Python "assert" statements on.'
-  echo ''
-  echo '  help'
-  echo '    Shows this description and exits.'
-  echo ''
-  echo '  interfaceCheck'
-  echo '    Do not attempt to run the model executable, but read from existing output files.'
+# if there's no file to run after parsing arguments, run "help" after warning of the problem.
+if [ ${#ARGS[@]} -eq 0 ]; then
   echo ''
-  echo '  interactive'
-  echo '    Run RAVEN in GUI mode if possible.'
-  echo ''
-  echo '  interactiveCheck'
-  echo '    Run tests on interactive GUI.'
-  echo ''
-elif [ "$MODE" = 'opt' ]; then
-  python -O $SCRIPT_DIR/framework/Driver.py "$ARGS"
-else
-  python $SCRIPT_DIR/framework/Driver.py "$ARGS"
+  echo 'ERROR: No input file specified! See options below.'
+  MODE='help'
 fi
 
+# run Driver.py based on the mode chosen
+case $MODE in
+  'opt' )
+    echo 'Running RAVEN in "opt" mode.'
+    python -O $SCRIPT_DIR/framework/Driver.py "$ARGS"
+    ;;
+  'dev' )
+    echo 'Running RAVEN in "dev" mode.'
+    python $SCRIPT_DIR/framework/Driver.py "$ARGS"
+    ;;
+  'help' )
+    echo ''
+    echo 'Usage:'
+    echo '  raven_framework [-D][interfaceCheck][interactive][interactiveCheck] file'
+    echo ''
+    echo 'Description:'
+    echo '  Runs RAVEN using the input file given by "file".'
+    echo ''
+    echo 'Options:'
+    echo '  -D'
+    echo '    Development mode.  Turns Python "assert" statements on.'
```

Comment:
add the description about the 'opt' mode

Reply:
I don't think so.  There is no flag or option for it, so it does not belong in the description of "help", I believe.

Reply:
For me it is more clear to add this description. Because if the user just run 'raven_framework' without the input file, the current 'help' message will guide the user to run the code in 'dev' mode instead of 'opt' mode. I think we should put 'opt' there, and make it default as the code does.

Reply:
any thoughts here?

Reply:
I don't feel like the help guides the user to run in dev mode, unless they feel like they want to run with assert statements on.  The usage syntax marks the default operation as being without any optional flags.  I'd rather remove the "help" than try and replicate the user manual in it; why do you feel the user is being guided towards dev mode?

Reply:
I increased the verbosity and added a "default usage" mode, which is somewhat unorthodox but might be more clear to users new to *nix.

Reply:
First, I agree with you we do not want to replicate the user manual in the code. But a brief clear description I think is also useful. Second, I do not think we have documented this new feature in the user manual (we may not need now since this changes only affect the developers), that's why I suggest to put it here. Anyway, my thoughts may be wrong, I really appreciate that you made the changes. To me it is more clear, however it may the have no difference to the users.